lib/cli/parse.rs

```diff
@@ -56,8 +65,12 @@ pub(super) fn secret_key_from_file<P: AsRef<Path>>(
 }
 
 pub(super) fn timestamp(value: &str) -> Result<Timestamp, CliError> {
+    #[cfg(any(feature = "std-fs-io", test))]
+    let timestamp = Timestamp::now();
+    #[cfg(not(any(feature = "std-fs-io", test)))]
+    let timestamp = Timestamp::zero();
```

Comment:
we should maybe not do any defaulting when the feature is off and just return an error? I think that might be less surprising

Reply:
No that the purpose we want to avoid an error in that case when you are not in an OS but in wasm, being able to pass the step of construction with a zero timestamp because there is no clock in this context and the consumer can add a timestamp later with it's own clock and set_timestamp

Reply:
we can just require them to pass the timestamp as an argument and only fail if they don't specify it

Reply:
Sorry I made a confusion with this PR https://github.com/casper-network/casper-node/pull/4714/files#diff-28b84608019640a79626bab09ac38b648c6c87950af0d1f48c5bcd908533803bR99 in some constructors where you can not pass a value at that moment my bad.

So for this timestamp in create_transaction()

```
    let timestamp = parse::timestamp(transaction_params.timestamp)?;
    ```
I think we should not error when people are building a transaction from wasm because they miss a timestamp that can't be produced in wasm if they don't pass a value, I thing the transaction should be with a zero timestamp and indeed they can add their time in transaction_params or they can call set_timestamp later but erroring here specifically for wasm for this timestamp is more confusing than letting it to zero without any indication in wasm.

Reply:
is there anything preventing the user from specifying the timestamp there? it seems odd to me to create it with a zero timestamp to just change it later

Reply:
There is nothing for the consumer point of view to check that he passes a timestamp but it's even more surprising to fail at that moment if you don't pass it as not really compulsory in normal context, but you are in wasm we can't timestamp for you at that moment

in wasm 
```
const transaction_params = new TransactionStrParams(chain_name, initiator_addr, secret_key);
```
it would not be very  intuitive to fail here because timestamp is not given, it can be ensured by the SDK behind the hood you are right as a consumer in the SDK I ensure that the timestamp is there in params 

https://github.com/casper-ecosystem/rustSDK/blob/dev/src/types/transaction_params/transaction_str_params.rs#L276-L278

however if in the future somebody is using this builder from the client library like the SDK thus only in wasm transaction_params would not error, it would "just" have a zero timestamp and then if put_transaction this would be rejected so he would have to add in his framework an auto timestamp from his wasm clock (chrono probably)